src/client.ts

```diff
@@ -8037,7 +8025,7 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
         kind: PushRuleKind,
         ruleId: Exclude<string, RuleId>,
         callback?: Callback,
-    ): Promise<any> { // TODO: Types
+    ): Promise<{}> {
```

Comment:
Mostly to show it's deliberate and not an attempt to evade types:
```suggestion
    ): Promise<{}> { // API returns an empty object
```

Reply:
JSDoc a few lines up says it is deliberate